static/js/components/NewAPIObservation.jsx

```diff
@@ -0,0 +1,154 @@
+
+dayjs.extend(utc);
+
+const NewAPIObservation = () => {
+  const { instrumentList } = useSelector((state) => state.instruments);
+  const { telescopeList } = useSelector((state) => state.telescopes);
+  const { allocationList } = useSelector((state) => state.allocations);
+  const allGroups = useSelector((state) => state.groups.all);
+
+  const dispatch = useDispatch();
+
+  const nowDate = dayjs().utc().format("YYYY-MM-DDTHH:mm:ssZ");
+  const defaultStartDate = dayjs()
+    .subtract(3, "day")
+    .utc()
+    .format("YYYY-MM-DDTHH:mm:ssZ");
+  const defaultEndDate = dayjs().utc().format("YYYY-MM-DDTHH:mm:ssZ");
+
+  useEffect(() => {
+    const getAllocations = async () => {
+      // Wait for the allocations to update before setting
+      // the new default form fields, so that the allocations list can
+      // update
+
+      await dispatch(
+        allocationActions.fetchAllocations({
+          apiType: "api_classname_obsplan",
+        })
+      );
+    };
+
+    getAllocations();
+
+    dispatch(
+      instrumentActions.fetchInstrumentForms({
+        apiType: "api_classname_obsplan",
+      })
+    );
+    dispatch(
+      allocationActions.fetchAllocations({
+        apiType: "api_classname_obsplan",
+      })
+    );
+
+    // Don't want to reset everytime the component rerenders and
+    // the defaultStartDate is updated, so ignore ESLint here
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [dispatch]);
+
+  if (
+    !allGroups ||
+    allGroups.length === 0 ||
+    telescopeList.length === 0 ||
+    instrumentList.length === 0 ||
+    allocationList.length === 0
+  ) {
+    return (
+      <div>
+        <CircularProgress color="secondary" />
+      </div>
+    );
+  }
+
+  const groupLookUp = {};
+  // eslint-disable-next-line no-unused-expressions
+  allGroups?.forEach((group) => {
+    groupLookUp[group.id] = group;
+  });
+
+  const telLookUp = {};
+  // eslint-disable-next-line no-unused-expressions
+  telescopeList?.forEach((tel) => {
+    telLookUp[tel.id] = tel;
+  });
+
+  const instLookUp = {};
+  // eslint-disable-next-line no-unused-expressions
+  instrumentList?.forEach((instrumentObj) => {
+    instLookUp[instrumentObj.id] = instrumentObj;
+  });
+
+  const handleSubmit = async ({ formData }) => {
+    await dispatch(observationActions.requestAPIObservations(formData));
+  };
+
+  function validate(formData, errors) {
+    if (nowDate < formData.end_date) {
+      errors.end_date.addError(
+        "End date must be before current time, please fix."
+      );
+    }
+    if (formData.start_date > formData.end_date) {
+      errors.start_date.addError(
+        "Start date must be before end date, please fix."
+      );
+    }
+    return errors;
+  }
+
+  const allocationFormSchema = {
+    type: "object",
+    properties: {
+      start_date: {
+        type: "string",
+        format: "date-time",
+        title: "Start Date (Local Time)",
```

Comment:
Just making sure: the dates given in local time are correctly translated to UTC in the DB?

Reply:
@Theodlz How does this stuff work exactly? Where does the conversion to UTC happen and what do I need to do to ensure that occurs?

Reply:
Added the +00:00.